### PR TITLE
WalletConnect show expired toast

### LINKED ISSUE #20758: WalletConnect show expired toast
If the user signs an expired request, there is no error toast. We should show an "expired" toast and shouldn't proceed signing the request.

**Steps:** 

1. Connect to dApp and trigger any request
2. Keep request modal open in Status app (do not sign the request)
3. Wait until request will be expired on dApp's side
4. See if Expired state is somehow reflected in Status app UI

**Actual result:** Expired state is not reflected. User can sign request which has been already expired, no error will be shown. 

React App 2024-07-15 15-05-28

photo_2024-07-15 15 16 41

**Expected result:** Show an error toast that tells the user the request has expired.

_Originally posted by @pavloburykh in https://github.com/status-im/status-mobile/issues/20693#issuecomment-2228366716_

### PR DESCRIPTION
**Pull request by alwx**
fixes #20758 

#### Platforms

- Android
- iOS

status: ready

### CODE REVIEW

**Review by clauxx — APPROVED** [review]
Looks good, but can we add the same for session proposals? I guess it would be the same issue there with expired requests.

**Comment by alwx** [comment]
@clauxx done!

File: src/status_im/contexts/wallet/wallet_connect/events.cljs [hunk 1]
```diff
@@ -153,21 +153,28 @@
...
+               :proposal             current-proposal
+               :supported-namespaces supported-namespaces
+               :on-success           (fn [approved-session]
+                                       (log/info "Wallet Connect session approved")
+                                       (rf/dispatch [:wallet-connect/reset-current-session-proposal])
+                                       (rf/dispatch [:wallet-connect/persist-session approved-session]))
+               :on-fail              (fn [error]
+                                       (log/error "Wallet Connect session approval failed"
+                                                  {:error error
+                                                   :event :wallet-connect/approve-session})
```
Comment (ilmotta) [inline src/status_im/contexts/wallet/wallet_connect/events.cljs]: Maybe you like this, maybe not, but we can get the event ID out of the coeffects, as we do here:

https://github.com/status-im/status-mobile/blob/e4639c153f4ff499fd4aa535fd92ad4291e6644e/src/status_im/contexts/communities/actions/addresses_for_permissions/events.cljs#L9

**Review by ilmotta — APPROVED** [review]

**Comment by qoqobolo** [comment]
Thanks for the PR @alwx. 
One question shouldn't the text and the icon be red?🤔 Now it looks like a "success notification"
Screenshot 2024-07-24 at 18 49 23

**Comment by qoqobolo** [comment]
> Thanks for the PR @alwx. One question shouldn't the text and the icon be red?🤔 Now it looks like a "success notification" Screenshot 2024-07-24 at 18 49 23

@alwx we can fix it separately if needed.
PR can be merged, thanks!

**Comment by clauxx** [comment]
> Thanks for the PR @alwx. One question shouldn't the text and the icon be red?🤔 Now it looks like a "success notification" Screenshot 2024-07-24 at 18 49 23

mm I guess should be a negative toast, but not a big problem I think.

**Comment by qoqobolo** [comment]
> ess should be a negative toast, but not a big problem I think.

I'll log it for further discussion

**Comment by alwx** [comment]
@qoqobolo I already fixed it, no need to spend time logging since it's a matter of one-word change (from `:positive` to `:negative`)

**Comment by qoqobolo** [comment]
> @qoqobolo I already fixed it, no need to spend time logging since it's a matter of one-word change (from `:positive` to `:negative`)

Thank you. Should I check it manually?

**Comment by alwx** [comment]
@qoqobolo there are very few things that might go wrong with it but you can check :) But I've also rebased it, and since there were some changes done to the component, might be better to just check if it still works as expected.

**Comment by qoqobolo** [comment]
@alwx looks good, let's merge it!

**Pull request merged**